SF iOS/SF iOSTests/EventDataSourceTests.m

```diff
@@ -7,40 +7,56 @@
 //
 
 #import <XCTest/XCTest.h>
-#import "EventDataSource.h"
-
-@interface EventDataSourceTests : XCTestCase
+#import "EventDataSource+RealmConfiguration.h"
+#import <Realm/Realm.h>
 
+@interface EventDataSourceTests : XCTestCase <EventDataSourceDelegate>
+@property (nonatomic) EventDataSource *dataSource;
+@property (nonatomic) XCTestExpectation *fetchingExpectation;
 @end
 
 @implementation EventDataSourceTests
 
 - (void)setUp {
+    _fetchingExpectation = [self expectationWithDescription:@"Wait for events"];
+
+    RLMRealmConfiguration *realmConfiguration = [RLMRealmConfiguration defaultConfiguration];
+    
+    // In-memory Realms do not save data across app launches, but all other features of Realm will work as expected, including querying, relationships and thread-safety. This is a useful option if you need flexible data access without the overhead of disk persistence.
+    [realmConfiguration setInMemoryIdentifier:@"UnitTest"];
+
+    [RLMRealmConfiguration setDefaultConfiguration:realmConfiguration];
+    
+    self.dataSource = [[EventDataSource alloc] initWithEventType:EventTypeSFCoffee withRealmConfiguration:realmConfiguration];
+    
+    self.dataSource.delegate = self;
+    
     [super setUp];
 }
 
+- (void)testFetchingCoffeeEvents {
+    XCTAssertEqual(self.dataSource.numberOfEvents, 0);
+    [self.dataSource refresh];
+
+    [self waitForExpectationsWithTimeout:10.0 handler:^(NSError *error) {
+        if (error) {
+            NSLog(@"%@",error.localizedDescription);
+        }
+    }];
+}
+
+// MARK: - EventDataSourceDelegate
+- (void)didChangeDataSourceWithInsertions:(nullable NSArray<NSIndexPath *> *)insertions updates:(nullable NSArray<NSIndexPath *> *)updates deletions:(nullable NSArray<NSIndexPath *> *)deletions {
+    [_fetchingExpectation fulfill];
+    // The hard-coded value is not great, but the endpoint URL is currently hard-coded 2019-04-22
+    XCTAssertGreaterThanOrEqual(self.dataSource.numberOfEvents, 70);
```

Comment:
And this assertion should actually be in the test itself rather than at the test suite class level. I'd just copy the assertion after line 44 above.